front-end/src/app/shared/components/inputs/additional-info-input/additional-info-input.component.ts

```diff
@@ -16,39 +16,44 @@ export class AdditionalInfoInputComponent extends BaseInputComponent implements
 
-    const purposeDescriptionPrefix = this.transaction?.transactionType?.purposeDescriptionPrefix;
-
-    if (purposeDescriptionPrefix) {
-      // Add custom prefix required validation function to purpose description
-      this.form.controls[this.templateMap.purpose_description].addValidators([
-        ValidateUtils.prefixRequiredValidator(purposeDescriptionPrefix),
-      ]);
-
-      // Watch changes to purpose description to make sure prefix is maintained
-      this.form
-        .get(this.templateMap.purpose_description)
-        ?.valueChanges.pipe(takeUntil(this.destroy$))
-        .subscribe((value: string) => {
-          if (value.length < purposeDescriptionPrefix.length) {
-            // Ensure prefix is the first part of the string in the textarea if no user text added
-            this.form.get(this.templateMap.purpose_description)?.setValue(purposeDescriptionPrefix);
-          } else if (!value.startsWith(purposeDescriptionPrefix)) {
-            // Retain user text in textarea if possible if user changes prefix
-            this.form
-              .get(this.templateMap.purpose_description)
-              ?.setValue(purposeDescriptionPrefix + value.slice(value.indexOf(': ') + 2));
-          }
-        });
-
-      // Initialize value of purpose description to prefix if empty
-      if (!this.form.get(this.templateMap.purpose_description)?.value) {
-        this.form.get(this.templateMap.purpose_description)?.setValue(purposeDescriptionPrefix);
-      }
+    if (this.transaction?.transactionType?.purposeDescriptionPrefix) {
+      this.initPrefix(
+        this.templateMap.purpose_description,
+        this.transaction?.transactionType?.purposeDescriptionPrefix
+      );
+    }
+
+    if (this.transaction?.transactionType?.memoTextPrefix) {
+      this.initPrefix(this.templateMap.text4000, this.transaction?.transactionType?.memoTextPrefix);
     }
   }
 
   isDescriptionSystemGenerated(): boolean {
     // Description is system generated if there is a defined function.  Otherwise, it's mutable
     return this.transaction?.transactionType?.generatePurposeDescription !== undefined;
   }
+
+  initPrefix(field: string, prefix: string) {
+    // Add custom prefix required validation function to form text field
+    this.form.controls[field].addValidators([ValidateUtils.prefixRequiredValidator(prefix)]);
+
+    // Watch changes to form text field to make sure prefix is maintained
+    this.form
+      .get(field)
+      ?.valueChanges.pipe(takeUntil(this.destroy$))
+      .subscribe((value: string) => {
+        if (value.length < prefix.length) {
+          // Ensure prefix is the first part of the string in the textarea if no user text added
+          this.form.get(field)?.setValue(prefix);
+        } else if (!value.startsWith(prefix)) {
+          // Retain user text in textarea if possible if user changes prefix
+          this.form.get(field)?.setValue(prefix + value.slice(value.indexOf(': ') + 2));
```

Comment:
One gotchya here is if they delete : character the UI has an issue.  It might be worth adding an OR to line 45 if to check for missing :